src/libxively/sft/xi_sft_revision.c

```diff
@@ -163,15 +163,27 @@ xi_state_t xi_sft_revision_firmware_ok()
     char* xi_firmware_resource_name = firmware_update_data;
     char* xi_firmware_revision      = firmware_update_data;
 
-    strtok_r( xi_firmware_revision, "\n", &xi_firmware_revision );
+    while ( '\n' != *xi_firmware_revision && 0 != *xi_firmware_revision )
+    {
+        ++xi_firmware_revision;
+    }
+
```

Comment:
Just making sure that you know that this string is always null terminated?  It makes me nervous that we don't put a cap on this increment loop based on the size of the string.

Reply:
Perhaps add an out parameter to xi_sft_revision_read_string_from_file that is the buffer size or the string length, so that you can do a limit on the number of bytes compared here.

Reply:
A string is string because a zero indicates its end. Otherwise it is a series of bytes. The function name xi_sft_revision_read_string_from_file says "...read_string..." so the output is a string which is null terminated by definition. Other solution is to rename the function and call it xi_sft_revision_read_data_from_file or ...read_bytes... this function must have a length output parameter.
So either of these is sufficient. I don't think two indicators should carry the content end information. Here I chose the terminating zero since I exactly know that the file content does not have zero since it is a string, the zero is the end.
So I wouldn't touch this only if you preferred the "...read_bytes..." function renaming. But this wouldn't make sense because this file contains a string, its purpose is to contain a string.

Reply:
I bring this up because I know that when people code for embedded devices, they put in safeguards to ensure they don't do buffer overruns.  things like strncmp and strncpy and strncat exist for these reasons.  

But now after looking closer I see that the buffer guarantees a null terminator in there.  So I'm ok with it.